### DISCUSSION TITLE
ORP Dosing Delay on Startup Possible?

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#932] [Q&A]
Is it possible to set an ORP dosing delay when the filter pump starts like we can for the pH dosing?  I changed the delay for pH to 5 minutes because it appears to take a little while for the probe readings to stabilize when the filter pump starts each morning.  The filter pump runs at 1750 RPM which is about 17 GPM.

While the pH delay was counting down I noticed the salt cell turned on for a second or two and then turned off.  This cycle repeated multiple times until the pH delay expired and the acid pump turned on (thereby turning off the salt cell until the pH dose completed).  By the time the relatively brief pH dose completed, the ORP reading had stabilized & no additional chlorine was needed.

I observed this behavior for the first time this morning (I changed the pH delay yesterday).  I did not have a capture running and because enough pool water has filled the flow cell, I cannot do so before tomorrow.  That being said,  I will try to capture the behavior tomorrow if the overnight probe readings drop sufficiently.  Of course, if there is a way to delay the initial chlorine dose, I'll use that option.

Thank you!

**Comment by tagyoureit** [comment]
A few thoughts.

First, the pH and ORP probes are designed to stay wet all the time.  If your flow cell is draining when the pool is off, I would add one (or maybe two) one way valves so the probes are never exposed to air.  You may still get a variance at startup but the introduction of air bubbles into the cell at startup is probably as much a culprit, if not more, than the actual water readings.

Second, in REM you can smooth out the readings over n number of samplings which would make the readings less erratic.

Finally, if the cell turns on or off for a few seconds here or there it isn't really much of a concern.  The start delays are more to make sure that flow has been established and we aren't pooling chemicals.  You can set this with pH and chlorine dispensers because they don't know about the flow.  The salt cells all have their own flow sensor so it isn't a concern as much.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/1646840/e6f0c2e2-6f77-49e8-8129-f10eb2a2d6b9)

**Reply by FlaMike** [reply]
Thank you for your suggestions!  I do not know definitively if the flow cell drains overnight.  I certainly hope not, of course & will check tomorrow morning before the pool turns on.  

Currently, there is a one way valve on the drain side of the flow cell, but it would not prevent the cell from draining, of course.  You recommended I install it because I am running the fill & drain lines from the two ports on the Intelliflow pump.  I will get and install a second one way valve for the fill side of the cell.  It certainly would prevent the cell from draining from the fill side when the the pump is not running.

I have observed some minor leakage of air/water around both probes at the flow cell.  I re-taped & reinstalled both to no avail.  I just got a 3/4"  brass NPT plug.  I'm planning to use it to potentially remove any burrs/irregularities on the cell's threads & hope I get a better seal.  (Somewhat like you recommended for flow switch cross-thread problem I experienced a few weeks ago.)  

I updated the REM settings so all feeds from both probes take 3 samples to smooth readings.  Previously, the ORP setting was 1; the pH was already set to 3.

Finally, it's good to know the salt cell won't be damaged by toggling on/off a few times when the pool starts up in the morning--the only time I've seen that behavior.

As always, I appreciate greatly your help &  guidance!

**Reply by FlaMike** [reply]
Small, but hopeful, update since my posting a little while ago:

I installed the brass NPT plug temporarily in each probe location (with the probes' caps filled with pool from the flow cell), removed the plug & re-installed the probes with a *ton* of teflon tape.  To my amazement, I was able to screw each probe between 1/8" and 1/4" further into the flow cell.  Each probe now has only a couple of threads that are not engaged in the flow cell.  First indications are there is no visible water "weeping" or air infiltration at either probe location.  Of course, it's a little hard to be certain because everything is wet from my messing around.  Nevertheless, I am (very) cautiously hopeful that everything is truly leak-free at this time.

As I recall, when I  installed the A-S pressure sensor in the sand filter, it took me a few tries & days before it stopped weeping water.  I was too afraid I would over-tighten it & crack something.  It hasn't missed a beat since I finally screwed it in tightly enough.  Hoping the same is true with the pH & ORP probes now!

I will check the water level in the flow cell in the morning before the pump starts.  Fingers crossed.

The second one way valve--for the supply side of the flow cell-should arrive Thursday & will be installed as shortly after its arrival.

**Reply by FlaMike** [reply]
I inspected the flow cell this morning before the filter pump started.  I'm using an IPS flow cell as recommended in the Wiki.  I think it has 3 chambers (inflow where the flow switch is located, main chamber (or 2?) where the pH and ORP probes are located, and outflow).  The inflow and outflow chambers did not appear to have water in them.  However, the main chamber(s) with the probes was filled with water & only had a few very small air bubbles.  There did not appear to be any air bubbles along or at the tips of the probes.

I started the filter pump & the inflow & outflow chambers filled with water and the flow switch float, er floated to the top of its cell & REM showed the switch had closed indicating sufficient water flow.  Per usual, the outflow chamber was not 100% filled once the pump slowed from priming, but the chamber was mostly filled & I could see the spillover flow from the main chamber.

There appeared to be a small weep around the ORP probe, but it did not appear to be significant and no air bubbles appeared to be entering the main chamber from either probe.

I just checked the flow cell again--the pool has been running for about 3 hours.  Everything looks good.  Inflow and main chambers are full and the outflow chamber is probably 90% or more full.  There are no air bubbles in the main chamber & the probes are bubble free & their readings are stable and reasonable.  REM dosing of acid & control of the salt cell appears to be normal.  The small weep at the ORP probe continues but does not appear to be an issue--no air appears to be infiltrating.

I will add the second one way valve to the inflow line when it arrives later in the week.  Once it's installed, I assume the inflow chamber should not evacuate when the filter pump turns off & air in the outflow chamber should not present a problem.

**Reply by tagyoureit** [reply]
It sounds like everything is ok as-is.  Not sure the 2nd 1-way valve will make a difference.  How varied were the readings this morning?

**Reply by FlaMike** [reply]
The readings were pretty good today--see below.  I'm not sure about the second one way valve as well, but I figured I'd replace the "extra" shutoff valve with it since it's already ordered.  Know anyone who wants extra John Guest valves :-)

One odd behavior with REM today.  We had a power failure (the dip in the graphs) & the pool controller rebooted when the whole house generator kicked in.  REM lost the pressure sensor setting & went back to using REM as a controller for the sand filter.  You may recall a couple of weeks ago, a capture I posted had errors regarding the sand filter that REM couldn't "turn on."  The errors reappeared when the power came back on & I had to re-administer the filter under "Bodies" to select "No server" for the connection.  Also the ORP readings in Home Assistant were messed up after utility power was restored--hence the flat part of the ORP graph.  (Both graphs are from HA.)  I did a hard reboot of Nixie, including flipping the breaker off for a few seconds, re-loaded the HA integration, & then ORP reporting in HA normalized.  REM's ORP reading was fine throughout.  Funky power surge stuff between the utility power & Generac?  I don't know.  All seemed good before &  after the power issues, though. 

A funny dip in ORP when I started the system this morning as you can see.  Not great magnitude in the change & it recovered quickly, so I wasn't too concerned.  If I should be, please let me know.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/59ab2c7c-4579-440a-a19d-48bfc93fea20)


![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/1e9ffee7-1568-4654-a94a-14c504148bb5)

**Reply by tagyoureit** [reply]
A bit of a spike/drop at the beginning (especially <+/-40) could easily be accounted for by stagnant water in the pipes sitting overnight, for example.  

Other than your utility company trying to screw with you, I don't see anything concerning here.  njsPC will adjust to any anomalous readings once they recover, and not that you have mostly normal readings there won't be any major impacts to the pool operation or chemical dosing.  

You might want to put a delay in the HA startup (if possible), so that it can connect to njsPC _after_ it's loaded up.  Or you could use the `npm run start:cached` to load njsPC without recompiling the TypeScript code (you'll just need to re-compile it when you pull) which might help it load before HA.  But again, I'm 👉 at the utility company for creating these issues.

**Reply by FlaMike** [reply]
The variation in probe values on start up do not concern me.  I understand the water in the flow cell & pipes is not representative of the pool water when it's been stagnant for hours on end.  In addition, the magnitude of the spike/dip is small compared to the "true" reading.  Finally, a little extra acid or chlorine at start up isn't significant, I agree.

When I first built Nixie I did try to use the cached flag.  However, I was not able to get it to work successfully with PM2.  If I recall correctly, and it's been a while, the system just hung & never started.  If you could point me to instructions on how to use the cached flag with PM2 correctly, I will be happy to make that modification.  I must admit, I get tired waiting for the system to download & rebuild when I'm at the pad & do a hard reboot.

Ain't it fun when the power company or folks doing construction on your mainline mess with you?

Thanks!!!

**Reply by tagyoureit** [reply]
[PM2 Wiki instructions](https://github.com/tagyoureit/nodejs-poolController/wiki/Automatically-start-at-boot---PM2-&-Systemd)

You would edit each app's startup:
```json
    {
      "name": "njsPC",
      "script": "npm",
      "args": [
        "run", // new
        "start:cached" // changed
      ],
```

This matches each applications `package.json` startup scripts:
```json
  "scripts": {
    "start": "npm run build && node dist/app.js",
    "start:cached": "node dist/app.js",
    "build": "tsc"
  },
```

Finally, you would run and save the new `ecosystem.config.js` per the initial instructions.
```
pm2 start ecosystem.config.js
pm2 save
```

**Reply by FlaMike** [reply]
Thank you for posting  the changes to the startup scripts.  In my earlier attempts to use the cache option  I don't think I had the "run" line you commented as the new line--I only did the "startL cached" line.  Probably why it didn't work.

It will likely be a couple of days until I can make the changes, but I will give them the old college try & be happy to report my success!

Thanks & have a super rest of the day!

**Reply by FlaMike** [reply]
OK, as it turned out I had a few minutes to play, so I made the cached suggestions you made to ecosystem.config.js file & went through the pm2 autostart setup process and it worked!!!  My earlier attempt did not work for because I did not inlcude the "run", instruction line.  Not sure if I just missed something in the documentation or it's missing in the pm2 Wiki, but all good now.

Thank you!!!!
